### PR TITLE
`etcdctl` precalculated argument values are validated before preset.

### PR DESCRIPTION
**Pull request by mimiteto**
**What this PR does / why we need it**:
`etcdctl` precalculated argument values are validated before preset.
If deemed inappropriate hinting message is printed.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix operator
`etcdctl` precalculated arguments are validated before set. If deemed inappropriate, hint is printed for the operator to set the correct values.
```

### CODE REVIEW

File: install_on_demand/.etcdctl [hunk 1]
```diff
@@ -36,16 +41,37 @@ function _etcdctl_prepend_args(){
   fi
 
   if _check_arg_present "--cert" "${args[@]}"; then
-    cert_args+=("--cert=/proc/$(get_etcd_pid)/root/var/etcd/ssl/client/client/tls.crt")
+    if [ -f "/proc/$(get_etcd_pid)/root/var/etcd/ssl/client/client/tls.crt" ]; then
+      cert_args+=("--cert=/proc/$(get_etcd_pid)/root/var/etcd/ssl/client/client/tls.crt")
+    else
+      echo "Certificate not found in the expected path." 1>&2
```
Comment (petersutter) [inline install_on_demand/.etcdctl:47]:
```suggestion
      echo "TLS certificate not found in the expected path." 1>&2
```

File: install_on_demand/.etcdctl [hunk 2]
```diff
@@ -36,16 +41,37 @@ function _etcdctl_prepend_args(){
...
+      echo -e "${red}You can use --cert=/path/to/file${nc}" 1>&2
+    fi
   fi
   if _check_arg_present "--key" "${args[@]}"; then
-    cert_args+=("--key=/proc/$(get_etcd_pid)/root/var/etcd/ssl/client/client/tls.key")
+    if [ -f "/proc/$(get_etcd_pid)/root/var/etcd/ssl/client/client/tls.key" ]; then
+      cert_args+=("--key=/proc/$(get_etcd_pid)/root/var/etcd/ssl/client/client/tls.key")
+    else
+      echo "Key not found in the expected path." 1>&2
```
Comment (petersutter) [inline install_on_demand/.etcdctl:55]:
```suggestion
      echo "TLS key not found in the expected path." 1>&2
```

File: install_on_demand/.etcdctl [hunk 3]
```diff
@@ -36,16 +41,37 @@ function _etcdctl_prepend_args(){
...
+      echo "CA certificate not found in the expected path." 1>&2
+      echo -e "${red}You can use --cacert=/path/to/file${nc}" 1>&2
+    fi
   fi
   if _check_arg_present "--endpoints" "${args[@]}"; then
-    cert_args+=("--endpoints=https://etcd-main-local:2379")
+    if nc -z -w "${nc_timeout}" "${etcd_host}" "${etcd_port}" 1>/dev/bull 2>/dev/null; then
+      cert_args+=("--endpoints=https://${etcd_host}:${etcd_port}")
+    else
+      echo "Could not connect to deffault etcd host - https://${etcd_host}:${etcd_port} for less than ${nc_timeout} seconds" 1>&2
```
Comment (petersutter) [inline install_on_demand/.etcdctl:71]:
```suggestion
      echo "etcd host https://${etcd_host}:${etcd_port} not reachable within ${nc_timeout} seconds" 1>&2
```

**Comment by petersutter** [comment]
Screenshot 2024-04-11 at 11 26 07
Some suggestions:
- Texts like `Certificate not found in the expected path.` should be displayed with warning color
- The suggestions like `You can use --cert=/path/to/file` should not display with error color, it's just a hint and should have the info color / no color
- maybe also print the path where the script was looking for the files

**Review by petersutter — APPROVED** [review]
not sure if I like the green color (for info) vs no color, but apart from that

/lgtm

**Comment by mimiteto** [comment]
It is info at the end of the day.
I don't have an opinion here.

**Review by plkokanov — APPROVED** [review]
/lgtm

**Pull request merged**